cmd/hostd/main.go

```diff
@@ -94,17 +95,46 @@ func openBrowser(url string) error {
 	}
 }
 
-// tryLoadConfig loads the config file specified by the HOSTD_CONFIG_FILE. If
-// the config file does not exist, it will not be loaded.
-func tryLoadConfig() {
-	configPath := "hostd.yml"
+func tryConfigPaths() []string {
 	if str := os.Getenv(configFileEnvVar); str != "" {
-		configPath = str
+		return []string{str}
 	}
 
-	if err := config.LoadFile(configPath, &cfg); err != nil && !errors.Is(err, os.ErrNotExist) {
-		checkFatalError("failed to load config file", err)
+	paths := []string{
+		"hostd.yml",
 	}
+	if str := os.Getenv(dataDirEnvVar); str != "" {
+		paths = append(paths, filepath.Join(str, "hostd.yml"))
+	}
+
+	switch runtime.GOOS {
+	case "windows":
+		paths = append(paths, filepath.Join(os.Getenv("APPDATA"), "hostd", "hostd.yml"))
+	case "darwin":
+		paths = append(paths, filepath.Join(os.Getenv("HOME"), "Library", "Application Support", "hostd", "hostd.yml"))
+	case "linux", "freebsd", "openbsd":
+		paths = append(paths,
+			filepath.Join(string(filepath.Separator), "etc", "hostd", "hostd.yml"),
+			filepath.Join(string(filepath.Separator), "var", "lib", "hostd", "hostd.yml"), // old default for the Linux service
+		)
+	}
+	return paths
+}
+
+// tryLoadConfig tries to load the config file. It will try multiple locations
+// based on GOOS starting with PWD/hostd.yml. If the file does not exist, it will
+// try the next location. If an error occurs while loading the file, it will
+// print the error and exit. If the config is successfully loaded, the path to
+// the config file is returned.
+func tryLoadConfig() string {
+	for _, fp := range tryConfigPaths() {
```

Comment:
Should we change the order of the paths in `tryConfigPaths` and have it return the OS specific paths first? Did you consider either logging what config file gets loaded or print a warning of multiple config files are present?

Reply:
I'm not sure about the ordering, but the goal was to be consistent in prioritizing most specific to least specific when deciding which setting to use. The priorities are as follows:

1. CLI Flags
2. Environment Variables
3. Config file
  3.1 PWD
  3.2 Data directory
  3.3 OS path